External/Plugins/HaXeContext/Context.cs

```diff
@@ -903,65 +903,143 @@ public override bool IsImported(MemberModel member, int atLine)
         /// <param name="cname">Class (short or full) name</param>
         /// <param name="inFile">Current file</param>
         /// <returns>A parsed class or an empty ClassModel if the class is not found</returns>
-        public override ClassModel ResolveType(string cname, FileModel inFile)
-        {
-            // unknown type
-            if (string.IsNullOrEmpty(cname) || cname == features.voidKey || classPath == null)
-                return ClassModel.VoidClass;
+        public override ClassModel ResolveType(string cname, FileModel inFile) => ResolveType(cname, inFile, true);
 
-            // handle generic types
-            if (cname.IndexOf('<') > 0)
+        public ClassModel ResolveType(string className, FileModel inFile, bool resolveStaticExtensions)
+        {
+            var result = ClassModel.VoidClass;
+            if (!string.IsNullOrEmpty(className) && className != features.voidKey && classPath != null)
```

Comment:
I prefer guarding cases to nesting many `if`s... Moreover the guarding change makes the diff difficult to follow.

I suppose the reason is that you wanted to lookup static extensions for the resolved type in all the cases. A better approach would have been to move the entire logic of `ResolveType` into an internal function, and apply the static extension resolution to the result.